daemon/daemon.go

```diff
@@ -61,21 +64,63 @@ func (ib *Infoblox) Allocate(args *ExtCmdArgs, result *types.Result) (err error)
 
 	mac := args.IfMac
 
-	log.Printf("RequestAddress: '%s', '%s', '%s'", netviewName, subnet, mac)
-	ip, _ := ib.Drv.RequestAddress(netviewName, subnet, "", mac, args.ContainerID)
+	return ib.requestAddress(conf, args, result, netviewName, subnet, mac)
+}
+
+func (ib *Infoblox) requestAddress(conf NetConfig, args *ExtCmdArgs, result *current.Result, netviewName string, cidr string, macAddr string) (err error) {
+
+	log.Printf("RequestAddress: '%s', '%s', '%s'", netviewName, cidr, macAddr)
+	ip, _ := ib.Drv.RequestAddress(netviewName, cidr, "", macAddr, args.ContainerID)
+
+	log.Printf("Allocated IP: '%s'", ip)
 
-	ipn, _ := types.ParseCIDR(subnet)
+	hwAddr, err := hwaddr.GenerateHardwareAddr4(net.ParseIP(ip), hwaddr.PrivateMACPrefix)
+	if err != nil {
+		log.Printf("Problem while generating hardware address using ip: %s", err)
+	}
+	_, err = ib.updateAddress(netviewName, cidr, ip, hwAddr.String())
+	if err != nil {
+		log.Printf("Problem while updating MacAddress: %s", err)
+	}
+	ipn, _ := types.ParseCIDR(cidr)
 	ipn.IP = net.ParseIP(ip)
-	result.IP4 = &types.IPConfig{
-		IP:      *ipn,
+	ipConfig := &current.IPConfig{
+		Version: "4",
+		Address: *ipn,
 		Gateway: conf.IPAM.Gateway,
-		Routes:  conf.IPAM.Routes,
 	}
+	routes := convertRoutesToCurrent(conf.IPAM.Routes)
+	result.IPs = []*current.IPConfig{ipConfig}
+	result.Routes = routes
 
 	log.Printf("Allocate result: '%s'", result)
 	return nil
 }
 
+func (ib *Infoblox) updateAddress(netviewName string, cidr string, ipAddr string, macAddr string) (string, error) {
```

Comment:
It is not useful at all if the first argument is always `""`. Please either 
- return the updated object as the first argument, or, 
- get rid of the first argument althogether